Migrate Azure.AI.ContentSafety to new Generator

radhgupta opened the pull request:
This pull request migrates Azure.AI.ContentSafety to new Generator.

Resolves: https://github.com/Azure/azure-sdk-for-net/issues/54491

jorgerangel-msft reviewed (commented):
There are some code checks that are failing https://dev.azure.com/azure-sdk/public/_build/results?buildId=5685364&view=logs&j=c3e74943-23a9-5e82-9b42-8286fce989d4&t=e049197e-d864-58b1-c0e9-38458fd26878. At a glance, we may need to re-export the API files.

jorgerangel-msft commented on sdk/contentsafety/Azure.AI.ContentSafety/src/Customizations/BlocklistClient.cs:
```diff
@@ -0,0 +1,57 @@
+        /// </param>
+        /// <param name="credential"> A credential used to authenticate to the service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public BlocklistClient(Uri endpoint, AzureKeyCredential credential) : this(endpoint?.AbsoluteUri, credential, new ContentSafetyClientOptions())
```
why do we need these custom ctors now?

jorgerangel-msft commented on sdk/contentsafety/Azure.AI.ContentSafety/src/Customizations/ContentSafetyClient.cs:
```diff
@@ -4,12 +4,59 @@
+        /// </param>
+        /// <param name="credential"> A credential used to authenticate to the service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public ContentSafetyClient(Uri endpoint, AzureKeyCredential credential) : this(endpoint?.AbsoluteUri, credential, new ContentSafetyClientOptions())
```
why do we need these custom ctors?

jorgerangel-msft commented on sdk/contentsafety/Azure.AI.ContentSafety/src/Customizations/ContentSafetyClientBuilderExtensions.cs:
```diff
@@ -1,13 +1,62 @@
+        public static IAzureClientBuilder<ContentSafetyClient, ContentSafetyClientOptions> AddContentSafetyClient<TBuilder>(this TBuilder builder, Uri endpoint)
+            where TBuilder : IAzureClientFactoryBuilderWithCredential
+        {
+            return builder.RegisterClientFactory<ContentSafetyClient, ContentSafetyClientOptions>((options, credential) => new ContentSafetyClient(endpoint, credential, options));
```
I'm wondering if we could reference the generated overload to avoid duplication here?

JonathanCrd commented on sdk/contentsafety/Azure.AI.ContentSafety/src/Customizations/ContentSafetyClientBuilderExtensions.cs:
```diff
@@ -1,13 +1,62 @@
 {
     /// <summary> ContentSafety client builder extensions. </summary>
     [CodeGenModel("AIContentSafetyClientBuilderExtensions")]
     public partial class ContentSafetyClientBuilderExtensions
```
Same here. It seems that before this file was only used for renaming the generated model.

JonathanCrd reviewed (commented):
Left a couple of comments. 

There are other task that are still needed for this PR:

1. There will be changes in the API, you have to regenerate them by running: 
```
eng\scripts\Export-API.ps1 contentsafety
```

2.  Test are failing due to "missing" recordings, but by looking at the CI logs this is actually a bug where the request URL is malformed with a duplicated path component, [here](https://dev.azure.com/azure-sdk/public/public%20Team/_build/results?buildId=5685364&view=logs&j=e5c983a4-22fd-5829-a047-e9c172755b46&t=5db61f33-424b-5adc-4c72-85f10d8f039f&l=1266).

radhgupta committed:
regen library

Update API references and clean up old config files after emitter migration

Remove accidentally committed local files (main.tsp and MIGRATION_COMPLETION_PLAN.md)

radhgupta committed:
updateQuery generated

radhgupta committed:
Add metadata.json file with API version 2024-09-01

radhgupta committed:
migraiton

radhgupta committed:
ignore test

radhgupta committed:
top renamed to maxCount

JoshLove-msft commented:
@radhgupta can you run this script and check in the updated file? https://github.com/Azure/azure-sdk-for-net/blob/main/doc/GeneratorMigration/Library_Inventory.ps1

radhgupta committed:
maxCount var fix

radhgupta committed:
added library inventory

radhgupta committed:
libary inventory

JoshLove-msft commented:
/azp run net - contentsafety

JoshLove-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into migrateContentSafety

JoshLove-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into migrateContentSafety

JoshLove-msft committed:
regen

JoshLove-msft committed:
fix attrs

JoshLove-msft committed:
usings
